scripts/actions/spells/black/drain.lua

```diff
@@ -30,20 +36,16 @@ spellObject.onSpellCast = function(caster, target, spell)
     dmg = addBonuses(caster, spell, target, dmg)
     --add in target adjustment
     dmg = adjustForTarget(target, dmg, spell:getElement())
-    --add in final adjustments
-    dmg = finalMagicAdjustments(caster, target, spell, dmg)
 
     if dmg < 0 then
         dmg = 0
     end
 
-    if target:getHP() < dmg then
-        dmg = target:getHP()
-    end
+    --add in final adjustments and deal damage
+    dmg = finalMagicAdjustments(caster, target, spell, dmg)
```

Comment:
er..crap. I just hit merge and didn't realize you have this after the check for negative dmg. Adjustments to the damage happen inside this function, so that's potentially problematic. Can you followup?

Reply:
but drain should never be negative, no? that's the point of that check in this function

Reply:
thats why we normally have the check for it being neagive coming after everything else. maybe finalMagicAdjustments has no chance of turning it negative - but I don't know that at a glance unless I have that function memorized.

Reply:
is there any instance of a mob being healed by drain? would that..unheal the player? At a minimum, I'd want to explain in comment in file.

Reply:
I believe vanilla custard in nyzul absorb dark. Without that check in LSB i imagine you'd drain it and damage yourself. Kinda funny, would be interested to know what happens in retail